types/src/byzantine.rs

```diff
@@ -346,13 +352,35 @@ pub(crate) mod test_utils {
         axis_idx: u16,
         axis: AxisType,
     ) -> BadEncodingFraudProof {
+        let mut rng = rand::thread_rng();
+
         let square_width = eds.square_width();
-        let mut nmt = eds.axis_nmt(axis, axis_idx).unwrap();
         let mut shares_with_proof: Vec<_> = Vec::with_capacity(square_width.into());
 
         // collect the shares for fraud proof
         for share_idx in 0..square_width {
-            let (share, proof) = nmt.get_index_with_proof(share_idx.into());
+            let proof_axis = match rng.gen_range(0..=1) {
+                0 => AxisType::Row,
+                1 => AxisType::Col,
+                _ => unreachable!(),
+            };
```

Comment:
we could do this, nit but doesn't need unreachable branch
```suggestion
            let proof_axis = if rand::random() {
                AxisType::Row
            } else {
                AxisType::Col
            };
```